File: problems-originals/0401-0500/0429_n-ary-tree-level-order-traversal/solution.js
/**
 * @param {Node} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
    if (root === null) return [];
    const levels = [];
    let level = [root];
    while (level.length > 0) {
        const values = [];
        const next = [];
        for (const node of level) {
            values.push(node.val);
            next.push(...node.children);
        }
        levels.push(values);
        level = next;
    }
    return levels;
};

File: problems-originals/0401-0500/0429_n-ary-tree-level-order-traversal/solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {

    public int[][] levelOrder(Node root) {
        List<int[]> levels = new ArrayList<>();
        if (root != null) {
            List<Node> level = new ArrayList<>();
            level.add(root);
            while (!level.isEmpty()) {
                int[] values = new int[level.size()];
                List<Node> next = new ArrayList<>();
                for (int index = 0; index < level.size(); index++) {
                    Node node = level.get(index);
                    values[index] = node.val;
                    next.addAll(node.children);
                }
                levels.add(values);
                level = next;
            }
        }
        return levels.toArray(new int[0][]);
    }
}
